mysql-test/suite/innodb/t/instant_alter_charset.test

```diff
@@ -0,0 +1,527 @@
+alter table all_binaries modify b text, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_binaries modify c mediumtext, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_binaries modify d longtext, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_binaries modify e varchar(150), algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_binaries modify f char(150), algorithm=instant;
+
+drop table all_binaries;
+
+create table all_strings (
+  a tinytext,
+  b text,
+  c mediumtext,
+  d longtext,
+  e varchar(150),
+  f char(150)
+) engine=innodb;
+
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_strings modify a tinyblob, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_strings modify b blob, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_strings modify c mediumblob, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_strings modify d longblob, algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_strings modify e varbinary(150), algorithm=instant;
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table all_strings modify f binary(150), algorithm=instant;
+
+drop table all_strings;
+
+create table key_part_change (
+  a char(150) charset ascii,
+  b char(150) charset ascii,
+  c char(150) charset ascii,
+  unique key ab (a,b)
+) engine=innodb;
+
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table key_part_change
+  modify a char(150) charset utf8mb4,
+  drop index ab,
+  add unique key ab(a,c),
+  algorithm=instant;
+
+drop table key_part_change;
+
+create table key_part_change_and_rename (
+  a char(100) charset ascii,
+  b char(100) charset ascii,
+  unique key ab (a,b)
+) engine=innodb;
+
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table key_part_change_and_rename
+  change a b char(100) charset utf8mb4,
+  change b a char(100) charset utf8mb4,
+  drop index ab,
+  add unique key ab(a,b),
+  algorithm=instant;
+
+drop table key_part_change_and_rename;
+
+create table enum_and_set (
+  a enum('one', 'two') charset utf8mb3,
+  b set('three', 'four') charset utf8mb3
+) engine=innodb;
+
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table enum_and_set
+  modify a enum('one', 'two') charset utf8mb4,
+  algorithm=instant;
+
+--error ER_ALTER_OPERATION_NOT_SUPPORTED_REASON
+alter table enum_and_set
+  modify b enum('three', 'four') charset utf8mb4,
+  algorithm=instant;
+
+drop table enum_and_set;
+
+create table compressed (
+  a varchar(255) charset utf8mb3 compressed
+) engine=innodb;
+
+insert into compressed values ('AAA'), ('bbb'), ('CCC');
+
+alter table compressed
+  modify a varchar(255) charset utf8mb4 compressed,
+  algorithm=instant;
+
+select * from compressed;
+check table compresed;
+
+drop table compressed;
```

Comment:
Could we preserve all tables where we did an `ALGORITHM=INSTANT` change until the end of the test?
At the end of the test:
1. Show that no `id` changed for the table or tablespace.
1. Restart the server.
1. Read all data from each table.
1. Drop the tables.